api-request-right-sizing.md

```diff
@@ -0,0 +1,87 @@
+# Container Request Right-Sizing API
+
+The container request right-sizing API provides recommendations for
+[container resource requests](https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/)
+based on configurable parameters and estimates the savings from implementing those recommendations
+on a per-container, per-controller level. Of course, if the cluster-level resources stay static then
+you will likely not enjoy real savings from applying these recommendations until you reduce
+your cluster resources. Instead, your idle allocation will increase.
+
+The endpoint is available at
+```
+http://<kubecost-address>/model/savings/requestSizing
+```
+
+## Parameters
+
+
+| Name | Type | Description |
+|------|------|-------------|
+| `targetCPUUtilization` | float between 0 and 1 | An amount of headroom to enforce with the new request, based on the calculated (real) usage. If the calculated usage is, for example, 100 mCPU and this parameter is `0.8`, the recommended CPU request will be `100 / 0.8 = 125` mCPU. |
+| `targetRAMUtilization` | float between 0 and 1 | Calculated like CPU. |
+| `window` | string | Duration of time over which to calculate usage. Supports hours or days before the current time in the following format: `2h` or `3d`. |
+| `algorithm` | string | The method with which to calculate usage. Currently supports either `max-headroom` or `percentile-headroom`. See subsections for additional parameters. If omitted, defaults to `percentile-headroom` for compatibility reasons. |
+
+### Max Headroom Algorithm
+
+The max headroom algorithm, available with `algorithm=max-headroom`, uses data in the Kubecost ETL to calculate the maximum resource usage (CPU and RAM) over the window. It requires no additional parameters.
+
+The max headroom algorithm is substantially more performant for large Kubernetes clusters because of the efficiency of Kubecost's ETL. Because recommendations are calculated at the controller level, and controllers can have many pods, we use a relatively straightforward algorithm to calculate maximum usage correctly. Here are some example cases:
+
+Weights (equivalent to a given Allocation's max CPU usage, for example) are on the left. Time moves right. Each "line" is the Allocation for one container, defined by the Allocation's Start and End.
+
+```
+3 pods, each with 1 container, under one deployment.
+3 | -----
+5 | -----
+7 | -----
+Correct max for the deployment: 3+5+7 = 15
+```
+
+```
+A CronJob that spins up 1 pod with 1 container.
+3 | ----
+5 |       -----
+7 |              -----
+Correct max for the CronJob: 7
+```
+
+```
+A CronJob that spins up 1 pod with 1 container on a cadence faster than it takes the
+job to run.
+3 | ------
+5 |    ------
+7 |        ------
+Correct max for the CronJob: 7+5 = 12
+```
```

Comment:
I don't feel super strongly about it, but I think all of this is implementation detail that doesn't necessarily need to be included here.